### DISCUSSION TITLE
[ideas] Should `tar_map` complain if `names` does not exist in `values`?

### DISCUSSION THREAD

**Discussion by KaiAragaki** [ropensci/tarchetypes#154] [Ideas]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

If a column given by `values` does not exist, random file names are given instead. Is this desired behavior, or would an error be appropriate?

``` r
library(targets)
tar_script({
    library(tarchetypes)
    values <- data.frame(names = c("A", "B"))
    list(tar_map(values = values, names = "i_dont_exist", tar_target(my_target, 
        "hi")))
})
tar_make()
#> ▶ start target my_target_6a0a0001
#> ● built target my_target_6a0a0001 [0.001 seconds]
#> ▶ start target my_target_df8a9eb9
#> ● built target my_target_df8a9eb9 [0 seconds]
#> ▶ end pipeline [0.035 seconds]
tar_manifest()
#> # A tibble: 2 × 2
#>   name               command 
#>   <chr>              <chr>   
#> 1 my_target_6a0a0001 "\"hi\""
#> 2 my_target_df8a9eb9 "\"hi\""
```

<sup>Created on 2023-08-08 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Thanks for making a beautiful pkg

**Comment by wlandau** [comment]
Unfortunately, that is not feasible because `tarchetypes` uses `tidyselect` for `names`. It is designed for calls like `tar_map(names = starts_with("my_target"))`, so `names` is automatically filtered to be a subset of `names(values)`.

**Reply by KaiAragaki** [reply]
Sorry, I'm not sure if I explained myself properly - or perhaps I'm misunderstanding you. In this proposal, I'm saying it SHOULD error if a column does not exist/no columns that satisfy `starts_with` exists, but currently it does not. Sorry if I'm just missing something here.

**Reply by wlandau** [reply]
I think throwing an error could lead to edge cases and friction in the general case. For example, suppose a user sets `names = -starts_with("bad_prefix")` to exclude certain names. I think they would expect `tar_map()` to work even if the suffixes become random hashes.

**Reply by wlandau** [reply]
As always, `tar_manifest()` and `tar_visnetwork()` are good ways to check that you have the names you want.
